src/graph.cpp

```diff
@@ -162,18 +166,24 @@ int Graph::addToTour(std::shared_ptr<Vertex> stop) {
  * @brief Iterates through the vertex set using DFS, respecting if an edge is selected or not
  * Time Complexity: O(|V|+|E|)
  * @param source - Vertex where the DFS starts
+ * @return execution errors (0 if none, -1 if couldn't calculate Edge length, -2 if self-loop)
  */
-void Graph::preorderMSTTraversal(std::shared_ptr<Vertex> source){
-/*    source->setVisited(true);
-    addToTour(source);
-    for (std::shared_ptr<Edge> &e : source->getAdj()){
-        if(!(e->getDest()->isVisited()) && e->isSelected()){
-            preorderMSTTraversal(e->getDest());
+int Graph::preorderMSTTraversal(std::shared_ptr<Vertex> source){
+    source->setVisited(true);
+    int exec_val = addToTour(source);
+    if (exec_val != 0) return exec_val;
+
+    for (size_t i = 0; i<vertexSet.size(); i++){
+        std::shared_ptr<Vertex> dest = findVertex(i);
+
+        if(!(dest->isVisited()) && selectedEdges[source->getId()][i]){
+            preorderMSTTraversal(dest);
         }
     }
 
-    if (source->getId() == 0) addToTour(source);
-*/}
+    if (source->getId() == 0) return addToTour(source);
```

Comment:
No, because its recursive, and by the time this function ends, all the "children" functions have already ended. So the node 0 only appears at the start and end of the tour

Reply:
Ok I get it makes sense, but I think the if check is a bit unnecessary then, why only do it when the source is 0?

Reply:
To close out the tour, so the first node in the tour will be at the end too (I can move it outside the function, but I think its cleaner this way (Even though it will run this check for every other node))

Reply:
Oh I get it now, because you don't wanna close out the tour with other source ids when it finishes running